resources/src/main/java/org/robolectric/res/android/Util.java

```diff
@@ -52,29 +54,34 @@ public static boolean isTruthy(Object o) {
     return o != null;
   }
 
-  static void ALOGD(String message, Object... args) {
+  @FormatMethod
+  static void ALOGD(@FormatString String message, Object... args) {
     if (DEBUG) {
       System.out.println("DEBUG: " + String.format(message, args));
     }
   }
 
-  static void ALOGW(String message, Object... args) {
+  @FormatMethod
+  static void ALOGW(@FormatString String message, Object... args) {
     System.out.println("WARN: " + String.format(message, args));
   }
 
-  public static void ALOGV(String message, Object... args) {
+  @FormatMethod
```

Comment:
Those code will not change frequently, do we need to include `errorprone` annotations to this files? If I understand correctly, the `AnnotateFormatMethod` warnings are raised by errorprone when building?

Reply:
Yes, we need the error annotation to eliminate the warning, this annotation helps to check if the pattern and value are compatible with each other when passed in the `String.format()`.

Yes, the `AnnotateFormatMethod` warning is raised by the building the project

Reply:
What I mean is related source code will not changed frequently, should we add `FormatMethod` after fixing found problems?

Reply:
sorry um.. what do we mean exactly here by `fixing found problems`?

Reply:
Oh, what I mean is there are some log formatting problem, such as %p not supported in Java, and we can fix those problems. The source code of this module will not change very frequently, and should we must include errorprone annotations to this module? If we don't include those annotations, will errorprone still raise warnings for us when building?

Reply:
Yes, if we don't include those the warnings will raise when building the project